src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeResource.kt

```diff
@@ -97,10 +99,36 @@ class BarcodeResource(private val barcodeService: BarcodeService, private val us
   ) = CheckBarcodeResponse(barcodeService.checkBarcode(userId, request.barcode, userContext.caseload))
 }
 
+data class CreateBarcodeRequest(
```

Comment:
This request body class looks a lot like `CreateContactRequest` - the differences are the swagger descriptions and the contact ID in this class.

Important difference though that I wanted to discuss is that `CreateContactRequest` in annotated with `@ContactHasDobOrPrisonNumber` to validate it has either a DOB or Prison Number.
The validation for this class should be the same, in that we want to validate the request body has either a DOB ot Prison Number, but I can't re-use `ContactHasDobOrPrisonNumber` (it is typed to `CreateContactRequest`) and I dont want to copy it.
Need to give this some thought .... 🤔

Reply:
I have deleted `ContactHasDobOrPrisonNumber` and replaced it with a simple function to validate the request body fields.